libs/external-db-dynamodb/src/dynamo_schema_provider.ts

```diff
@@ -86,21 +88,41 @@ export default class SchemaProvider implements ISchemaProvider {
 
         const { fields } = await this.collectionDataFor(collectionName)
 
-        if (!fields.some((f: { name: any }) => f.name === columnName)) {
-            throw new FieldDoesNotExist('Collection does not contain a field with this name')
+        if (!fields.some((f) => f.name === columnName)) {
+            throw new FieldDoesNotExist('Collection does not contain a field with this name', collectionName, columnName)
         }
         await this.docClient
-                  .update(dynamoRequests.removeColumnExpression(collectionName, fields.filter((f: { name: any }) => f.name !== columnName)))
+                  .update(dynamoRequests.updateColumnsExpression(collectionName, fields.filter((f: { name: any }) => f.name !== columnName)))
 
     }
 
-    async describeCollection(collectionName: string): Promise<ResponseField[]> {
+    async describeCollection(collectionName: string): Promise<Table> {
         await this.ensureSystemTableExists()
         validateTable(collectionName)
         
         const collection = await this.collectionDataFor(collectionName)
 
-        return [...SystemFields, ...collection.fields].map( reformatFields )
+        return {
+            id: collectionName,
+            fields: [...SystemFields, ...collection.fields].map(this.appendAdditionalRowDetails),
+            capabilities: this.collectionCapabilities()
+        }
+    }
+
+    async changeColumnType(collectionName: string, column: InputField): Promise<void> {
+        await this.ensureSystemTableExists()
+        validateTable(collectionName)
+        await validateSystemFields(column.name)
+        
+        const { fields } = await this.collectionDataFor(collectionName)
```

Comment:
What happens in the case of a request to change a column in a table that does not exist?

Reply:
`collectionDataFor` will throw `CollectionDoesNotExists`  error

btw, there's no scenario that could happen, because we are calling `changeColumnType` only if the collection exists and one of the column types have changed. 
if the collection is not exists, we'll create new one and won't call `changeColumnType`.